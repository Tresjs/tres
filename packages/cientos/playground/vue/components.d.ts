/* eslint-disable */
// @ts-nocheck
// Generated by unplugin-vue-components
// Read more: https://github.com/vuejs/core/pull/3399
// biome-ignore lint: disable
export {}

/* prettier-ignore */
declare module 'vue' {
  export interface GlobalComponents {
    AkuAku: typeof import('./src/components/gltf/BlenderCube.vue')['default']
    BlenderCube: typeof import('./src/components/gltf/BlenderCube.vue')['default']
    FboCube: typeof import('./src/components/FboCube.vue')['default']
    Gltf: typeof import('./src/components/gltf/index.vue')['default']
    GraphPane: typeof import('./src/components/GraphPane.vue')['default']
    Jeep: typeof import('./src/components/fbx/Jeep.vue')['default']
    ModelsDemo: typeof import('./src/components/ModelsDemo.vue')['default']
    OverlayInfo: typeof import('./src/components/OverlayInfo.vue')['default']
    RouterLink: typeof import('vue-router')['RouterLink']
    RouterView: typeof import('vue-router')['RouterView']
    StarsForTest: typeof import('./src/components/StarsForTest.vue')['default']
  }
}
