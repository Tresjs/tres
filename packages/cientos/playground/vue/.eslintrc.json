{
  "extends": "@tresjs/eslint-config-vue",
  "rules": {
    "no-console": "off",
    "vue/attribute-hyphenation": [2, "always", {
      "ignore": [
        "shadow-mapSize-width",
        "shadow-mapSize-height",
        "distortionMap",
        "material-uniforms-mieCoefficient-value",
        "material-uniforms-mieDirectionalG-value",
        "material-uniforms-sunPosition-value"
      ]
    }]
  }
}
