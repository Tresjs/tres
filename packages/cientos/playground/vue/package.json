{
  "name": "playground",
  "type": "module",
  "version": "0.0.0",
  "private": true,
  "scripts": {
    "dev": "vite --host",
    "build": "vue-tsc && vite build",
    "preview": "vite preview"
  },
  "dependencies": {
    "@tresjs/core": "5.0.0",
    "vue-router": "^4.5.1"
  },
  "devDependencies": {
    "@tresjs/leches": "^0.14.1",
    "unplugin-auto-import": "^20.1.0",
    "unplugin-vue-components": "^29.0.0",
    "vite-plugin-glsl": "^1.5.1",
    "vite-plugin-qrcode": "^0.3.0",
    "vue-tsc": "^3.0.7"
  }
}
