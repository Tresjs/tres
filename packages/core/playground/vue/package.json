{
  "name": "playground",
  "type": "module",
  "version": "0.0.0",
  "private": true,
  "scripts": {
    "dev": "vite --host",
    "build": "vite build",
    "preview": "vite preview" 
  },
  "dependencies": {
    "@tresjs/cientos": "5.0.0",
    "@tresjs/core": "workspace:^",
    "@tresjs/leches": "https://pkg.pr.new/@tresjs/leches@b34e795",
    "vue-router": "^4.5.1"
  },
  "devDependencies": {
    "@tweakpane/plugin-essentials": "^0.2.1",
    "unplugin-auto-import": "^20.2.0",
    "vite-plugin-glsl": "^1.5.1",
    "vite-plugin-qrcode": "^0.3.0",
    "vite-plugin-vue-devtools": "8.0.2",
    "vue-tsc": "^3.0.8"
  }
}
