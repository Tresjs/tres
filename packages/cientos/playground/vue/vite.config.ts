import { templateCompilerOptions } from '@tresjs/core'
import vue from '@vitejs/plugin-vue'
import { resolve } from 'pathe'
import UnoCSS from 'unocss/vite'
import AutoImport from 'unplugin-auto-import/vite'
import Components from 'unplugin-vue-components/vite'
import { defineConfig } from 'vite'
import glsl from 'vite-plugin-glsl'
import { qrcode } from 'vite-plugin-qrcode'

// https://vitejs.dev/config/
export default defineConfig({
  plugins: [
    glsl(),
    vue(templateCompilerOptions),
    AutoImport({
      dts: true,
      eslintrc: {
        enabled: true, // <-- this
      },
      imports: ['vue'],
    }),
    Components({
      /* options */
    }),
    UnoCSS({
      theme: {
        colors: {
          'cientos-blue': '#82dbc5',
          'cientos-orange': '#fbb03b',
        },
      },
    }),
    qrcode(),
  ],
  resolve: {
    alias: {
      '@tresjs/cientos': resolve(__dirname, '../../src/'),
    },
  },
})
