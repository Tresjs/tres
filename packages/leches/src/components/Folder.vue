<script setup lang="ts">
import { ref } from 'vue'
import type { Control } from '../types'
import ControlInput from './ControlInput.vue'

defineProps<{
  controls: Control[]
  label: string | number
}>()

function onChange(value: string, control: Control) {
  control.value = value as any
}

const isOpen = ref(false)

const toggle = () => {
  isOpen.value = !isOpen.value
}
</script>

<template>
  <div class="tl-mb-4">
    <button
      class="
        tl-flex
        tl-items-center
        tl-justify-between
        tl-w-full
        tl-py-2
        tl-px-4
        tl-bg-gray-100
        tl-border-none
        tl-text-gray-400
        tl-font-bold
        tl-text-xs
        tl-font-sans
        tl-cursor-pointer
      "
      :aria-expanded="isOpen"
      aria-haspopup="true"
      role="button"
      :data-folder="label"
      tabindex="0"
      @click="toggle"
    >
      <span>{{ label }}</span>
      <i :class="isOpen ? 'i-ic:baseline-keyboard-arrow-up' : 'i-ic:baseline-keyboard-arrow-down'"></i>
    </button>

    <Transition
      name="slide"
      enter-active-class="tl-animate-fade-in tl-animate-duration-200 tl-animate-ease-in-out"
      leave-active-class="tl-animate-fade-out tl-animate-duration-200 tl-animate-ease-in-out"
    >
      <div
        v-show="isOpen"
        class="tl-bg-white tl-rounded-b tl-pt-4"
        role="menu"
      >
        <template
          v-for="subcontrol in controls"
          :key="subcontrol.label"
        >
          <ControlInput
            :control="subcontrol"
            role="menuitem"
            @change="newValue => onChange(newValue, subcontrol)"
          />
        </template>
      </div>
    </Transition>
  </div>
</template>
