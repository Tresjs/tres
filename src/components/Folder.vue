<script setup lang="ts">
import { ref } from 'vue'
import type { LechesControl } from '../types'
import ControlInput from './ControlInput.vue'

defineProps<{
  controls: LechesControl[]
  label: string | number
}>()

const emit = defineEmits(['open'])

function onChange(value: string, control: LechesControl) {
  control.value = value as any
}

const isOpen = ref(false)

const toggle = () => {
  isOpen.value = !isOpen.value
  emit('open', isOpen.value)
}
</script>

<template>
  <div class="tl-mb-2 tl-transition-all tl-duration-400 tl-ease-in-out">
    <button
      class="
        tl-flex
        tl-items-center
        tl-justify-between
        tl-w-full
        tl-py-2
        tl-px-4
        tl-bg-gray-100
        dark:tl-bg-dark-300
        tl-border-none
        tl-text-gray-400
        dark:tl-text-gray-400
        tl-font-bold
        tl-text-xs
        tl-font-sans
        tl-cursor-pointer
        tl-relative
        tl-z-10
      "
      :aria-expanded="isOpen"
      aria-haspopup="true"
      role="button"
      :data-folder="label"
      tabindex="0"
      @click="toggle"
    >
      <span>{{ label }}</span>
      <i :class="isOpen ? 'i-ic:baseline-keyboard-arrow-up' : 'i-ic:baseline-keyboard-arrow-down'"></i>
    </button>

    <div class="tl-relative tl-overflow-hidden">
      <Transition
        name="slide"
        enter-active-class="tl-transition-[transform,opacity] tl-duration-200 tl-ease-out"
        enter-from-class="tl-origin-top tl-scale-y-0 tl-opacity-0"
        enter-to-class="tl-origin-top tl-scale-y-100 tl-opacity-100"
        leave-active-class="tl-transition-[transform,opacity] tl-duration-200 tl-ease-in"
        leave-from-class="tl-origin-top tl-scale-y-100 tl-opacity-100"
        leave-to-class="tl-origin-top tl-scale-y-0 tl-opacity-0"
      >
        <div
          v-show="isOpen"
          class="tl-bg-white dark:tl-bg-dark-300 tl-rounded-b tl-pt-4 tl-pb-2"
          role="menu"
        >
          <template
            v-for="subcontrol in controls"
            :key="subcontrol.label"
          >
            <ControlInput
              :control="subcontrol"
              role="menuitem"
              @change="newValue => onChange(newValue, subcontrol)"
            />
          </template>
        </div>
      </Transition>
    </div>
  </div>
</template>
