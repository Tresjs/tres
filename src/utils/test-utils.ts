import { createApp } from 'vue'
import type { Fn } from '@vueuse/core'

export function withSetup(composable: Fn) {
  let result
  const app = createApp({
    setup() {
      result = composable()
      // suppress missing template warning
      return () => null
    },
  })
  app.mount(document.createElement('div'))
  // return the result and the app instance
  // for testing provide / unmount
  return [result, app]
}
