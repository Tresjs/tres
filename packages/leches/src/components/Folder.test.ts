import { mount } from '@vue/test-utils'
import { TresLeches, useControls } from '/@/'
import { describe, expect, it } from 'vitest'
import { defineComponent } from 'vue'
import Folder from './Folder.vue'

class Vector3 {
  constructor(x = 0, y = 0, z = 0) {
    Vector3.prototype.isVector3 = true

    this.x = x
    this.y = y
    this.z = z
  }

  set(x, y, z) {
    if (z === undefined) { z = this.z } // sprite.scale.set(x,y)

    this.x = x
    this.y = y
    this.z = z

    return this
  }

  *[Symbol.iterator]() {
    yield this.x
    yield this.y
    yield this.z
  }
}

describe('folder Controls', () => {
  let wrapper
  let component
  const mountComponent = (setup) => {
    component = defineComponent({
      template: `<TresLeches :float="false" />`,
      components: { TresLeches, Folder },
      setup,
    })

    wrapper = mount(component, {
      components: { TresLeches, Folder },
    })
  }

  it('should render controls within a folder', () => {
    mountComponent(() => {
      const { cameraPosition } = useControls('camera', { position: new Vector3(3, 2, 4) })
      return { cameraPosition }
    })
    expect(wrapper.html().replace(/style="[^"]*"/, '')).toMatchSnapshot()
    expect(wrapper.find('button[data-folder="camera"]').exists()).toBe(true)
    expect(wrapper.find('input#default-cameraPosition-x').exists()).toBe(true)
    expect(wrapper.find('input#default-cameraPosition-y').exists()).toBe(true)
    expect(wrapper.find('input#default-cameraPosition-z').exists()).toBe(true)
  })

  it('should have unique keys for controls in different folders', () => {
    mountComponent(() => {
      const { cameraPosition } = useControls('camera', { position: new Vector3(3, 2, 4) })
      const { lightPosition } = useControls('light', { position: new Vector3(4, 4, 4) })
      return { controls: [cameraPosition, lightPosition], label: 'camera' }
    })
    const cameraInput = wrapper.find('input#default-cameraPosition-x')
    const lightInput = wrapper.find('input#default-lightPosition-x')
    expect(cameraInput.exists()).toBe(true)
    expect(lightInput.exists()).toBe(true)
    expect(cameraInput.element.value).toBe('3.00')
    expect(lightInput.element.value).toBe('4.00')
  })
})
