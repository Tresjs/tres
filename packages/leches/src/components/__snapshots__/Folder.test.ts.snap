// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Folder Controls > should render controls within a folder 1`] = `
"<div style=\\"left: 704px; top: 10px;\\" class=\\"tl-absolute tl-select-none tl-z-24 tl-w-280px tl-font-sans tl-text-xs\\">
  <div tabindex=\\"0\\" class=\\"tl-bg-white tl-shadow-xl tl-rounded tl-border-4 tl-border-solid tl-border-black\\">
    <header class=\\"tl-relative tl-cursor-grabbing tl-p-4 tl-flex tl-justify-between tl-text-gray-200 tl-relative\\"><i class=\\"tl-h-4 tl-w-4 tl-p-1.5 tl-flex tl-items-center tl-line-height-0 tl-rounded-full tl-bg-gray-100 tl-text-xs\\">üç∞</i>
      <div><i class=\\"i-ic-baseline-drag-indicator\\"></i><i class=\\"i-ic-baseline-drag-indicator\\"></i><i class=\\"i-ic-baseline-drag-indicator\\"></i></div>
      <div></div>
    </header>
    <div class=\\"tl-mb-4\\"><button class=\\"tl-flex tl-items-center tl-justify-between tl-w-full tl-py-2 tl-px-4 tl-bg-gray-100 tl-border-none tl-text-gray-400 tl-font-bold tl-text-xs tl-font-sans tl-cursor-pointer\\" aria-expanded=\\"false\\" aria-haspopup=\\"true\\" role=\\"button\\" data-folder=\\"camera\\" tabindex=\\"0\\"><span>camera</span><i class=\\"i-ic:baseline-keyboard-arrow-down\\"></i></button>
      <transition-stub name=\\"slide\\" enteractiveclass=\\"tl-animate-fade-in tl-animate-duration-200 tl-animate-ease-in-out\\" leaveactiveclass=\\"tl-animate-fade-out tl-animate-duration-200 tl-animate-ease-in-out\\" appear=\\"false\\" persisted=\\"true\\" css=\\"true\\">
        <div class=\\"tl-bg-white tl-rounded-b tl-pt-4\\" role=\\"menu\\" style=\\"display: none;\\">
          <div role=\\"menuitem\\">
            <div class=\\"tl-flex tl-px-4 tl-justify-between tl-gap-1 tl-items-center tl-mb-2\\"><label for=\\"cameraPosition\\" title=\\"position\\" class=\\"tl-text-gray-500 tl-w-1/3 tl-truncate\\">position</label>
              <div class=\\"tl-relative tl-w-2/3 tl-flex tl-justify-between tl-gap-0.5\\">
                <div class=\\"tl-flex tl-items-center tl-bg-gray-100 tl-rounded tl-w-1/3 tl-w-1/2\\"><span class=\\"tl-font-bold tl-px-1 tl-py-1 tl-text-0.65rem tl-text-gray-300\\">x</span><input id=\\"cameraPosition-x\\" type=\\"number\\" step=\\"1\\" class=\\"tl-w-full tl-px-0 tl-p-1 tl-text-right tl-text-0.65rem tl-text-gray-400 tl-bg-transparent focus:tl-border-gray-200 tl-outline-none tl-border-none tl-font-sans tl-appearence-none\\"></div>
                <div class=\\"tl-flex tl-items-center tl-bg-gray-100 tl-rounded tl-w-1/3 tl-w-1/2\\"><span class=\\"tl-font-bold tl-px-1 tl-py-1 tl-text-0.65rem tl-text-gray-300\\">y</span><input id=\\"cameraPosition-y\\" type=\\"number\\" step=\\"1\\" class=\\"tl-w-full tl-px-0 tl-p-1 tl-text-right tl-text-0.65rem tl-text-gray-400 tl-bg-transparent focus:tl-border-gray-200 tl-outline-none tl-border-none tl-font-sans tl-appearence-none\\"></div>
                <div class=\\"tl-flex tl-items-center tl-bg-gray-100 tl-rounded tl-w-1/3 tl-w-1/2\\"><span class=\\"tl-font-bold tl-px-1 tl-py-1 tl-text-0.65rem tl-text-gray-300\\">z</span><input id=\\"cameraPosition-z\\" type=\\"number\\" step=\\"1\\" class=\\"tl-w-full tl-px-0 tl-p-1 tl-text-right tl-text-0.65rem tl-text-gray-400 tl-bg-transparent focus:tl-border-gray-200 tl-outline-none tl-border-none tl-font-sans tl-appearence-none\\"></div>
              </div>
            </div>
          </div>
        </div>
      </transition-stub>
    </div>
    <!--v-if-->
  </div>
</div>"
`;

exports[`folder Controls > should render controls within a folder 1`] = `
"<div style="left: 704px; top: 10px;" class="tl-absolute tl-select-none tl-z-24 tl-w-280px tl-font-sans tl-text-xs">
  <div tabindex="0" class="tl-bg-white tl-shadow-xl tl-rounded tl-border-4 tl-border-solid tl-border-black">
    <header class="tl-relative tl-cursor-grabbing tl-p-4 tl-flex tl-justify-between tl-text-gray-200 tl-relative"><i class="tl-h-4 tl-w-4 tl-p-1.5 tl-flex tl-items-center tl-line-height-0 tl-rounded-full tl-bg-gray-100 tl-text-xs">üç∞</i>
      <div><i class="i-ic-baseline-drag-indicator"></i><i class="i-ic-baseline-drag-indicator"></i><i class="i-ic-baseline-drag-indicator"></i></div>
      <div></div>
    </header>
    <div class="tl-mb-4"><button class="tl-flex tl-items-center tl-justify-between tl-w-full tl-py-2 tl-px-4 tl-bg-gray-100 tl-border-none tl-text-gray-400 tl-font-bold tl-text-xs tl-font-sans tl-cursor-pointer" aria-expanded="false" aria-haspopup="true" role="button" data-folder="camera" tabindex="0"><span>camera</span><i class="i-ic:baseline-keyboard-arrow-down"></i></button>
      <transition-stub name="slide" enteractiveclass="tl-animate-fade-in tl-animate-duration-200 tl-animate-ease-in-out" leaveactiveclass="tl-animate-fade-out tl-animate-duration-200 tl-animate-ease-in-out" appear="false" persisted="true" css="true">
        <div class="tl-bg-white tl-rounded-b tl-pt-4" role="menu" style="display: none;">
          <div role="menuitem">
            <div class="tl-flex tl-px-4 tl-justify-between tl-gap-1 tl-items-center tl-mb-2"><label for="cameraPosition" title="position" class="tl-text-gray-500 tl-w-1/3 tl-truncate">position</label>
              <div class="tl-relative tl-w-2/3 tl-flex tl-justify-between tl-gap-0.5">
                <div class="tl-flex tl-items-center tl-bg-gray-100 tl-rounded tl-w-1/3 tl-w-1/2"><span class="tl-font-bold tl-px-1 tl-py-1 tl-text-0.65rem tl-text-gray-300">x</span><input id="cameraPosition-x" type="number" step="1" class="tl-w-full tl-px-0 tl-p-1 tl-text-right tl-text-0.65rem tl-text-gray-400 tl-bg-transparent focus:tl-border-gray-200 tl-outline-none tl-border-none tl-font-sans tl-appearence-none" value="3.00"></div>
                <div class="tl-flex tl-items-center tl-bg-gray-100 tl-rounded tl-w-1/3 tl-w-1/2"><span class="tl-font-bold tl-px-1 tl-py-1 tl-text-0.65rem tl-text-gray-300">y</span><input id="cameraPosition-y" type="number" step="1" class="tl-w-full tl-px-0 tl-p-1 tl-text-right tl-text-0.65rem tl-text-gray-400 tl-bg-transparent focus:tl-border-gray-200 tl-outline-none tl-border-none tl-font-sans tl-appearence-none" value="2.00"></div>
                <div class="tl-flex tl-items-center tl-bg-gray-100 tl-rounded tl-w-1/3 tl-w-1/2"><span class="tl-font-bold tl-px-1 tl-py-1 tl-text-0.65rem tl-text-gray-300">z</span><input id="cameraPosition-z" type="number" step="1" class="tl-w-full tl-px-0 tl-p-1 tl-text-right tl-text-0.65rem tl-text-gray-400 tl-bg-transparent focus:tl-border-gray-200 tl-outline-none tl-border-none tl-font-sans tl-appearence-none" value="4.00"></div>
              </div>
            </div>
          </div>
        </div>
      </transition-stub>
    </div>
    <!--v-if-->
  </div>
</div>"
`;
